### PR TITLE
Redis Stream Source Unit + E2E Tests

### PR DESCRIPTION
**Pull request by aavarghese**
Included in this PR:

- Unit tests for stream source api, lifecycle test, adapter, receive adapter etc
- Vendor dir updates (after updating K8 deps to v18.8)
- Code updates (after updating K8 deps to v18.8)
- Setup for github actions
- Adding test dir

### CODE REVIEW

**Comment by aavarghese** [comment]
https://github.com/aavarghese/eventing-redis/actions?query=workflow%3A%22Kind+e2e+tests%22

Workflow jobs are acting strange. I'm investigating...

**Comment by aavarghese** [comment]
cc @lionelvillard this set ready for review. I don't have an e2e test functional yet but other than that, test setup version 1 is in this PR. Can you take a look and let me know what you think so far?

File: hack/update-deps.sh [hunk 1]
```diff
@@ -54,11 +55,8 @@ fi
...
 
-rm -rf $(find vendor/ -name 'OWNERS')
-rm -rf $(find vendor/ -name 'OWNERS_ALIASES')
-rm -rf $(find vendor/ -name 'BUILD')
-rm -rf $(find vendor/ -name 'BUILD.bazel')
-rm -rf $(find vendor/ -name '*_test.go')
+find vendor/ -name 'OWNERS' -delete
+find vendor/ -name '*_test.go'-delete
 
 export GOFLAGS=-mod=vendor
```
Comment (lionelvillard) [inline hack/update-deps.sh:62]: The proper way now is to use `go_update_deps`: https://github.com/knative/eventing/blob/master/hack/update-deps.sh

File: test/OWNERS [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# The OWNERS file is used by prow to automatically merge approved PRs.
+
+approvers:
+- productivity-approvers
+
+reviewers:
+- productivity-reviewers
+
+labels:
+- area/test-and-release
```
Comment (lionelvillard) [inline test/OWNERS:10]: in order for this to work you need to add `productivity-approvers` and `productivity-reviewers` to `./OWNER_ALIASES`. See example here: https://github.com/knative/eventing/blob/master/OWNERS_ALIASES

**Pull request merged**